src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -426,100 +432,103 @@ public void acceptRecord(final SAMRecord record) {
+                    for (final Interval bait : probes) {
+                        for (final AlignmentBlock block : record.getAlignmentBlocks()) {
+                            final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
 
-            // Find the target overlaps
-            if (targets != null && !targets.isEmpty()) {
-                for (final Interval target : targets) {
-                    final Coverage coverage = this.coverageByTarget.get(target);
-
-                    for (final AlignmentBlock block : rec.getAlignmentBlocks()) {
-                        final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
-                        for (int pos=block.getReferenceStart(), readPos = block.getReadStart();
-                             pos<=end && pos >= target.getStart() && pos <= target.getEnd() && minimumBaseQuality <= baseQualities[readPos-1];
-                             ++pos, ++readPos) {
-                            ++this.metrics.ON_TARGET_BASES;
-                            if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
-                            coverage.addBase(pos - target.getStart());
+                            for (int pos=block.getReferenceStart(); pos<=end; ++pos) {
+                                if (pos >= bait.getStart() && pos <= bait.getEnd()) ++onBaitBases;
+                            }
                         }
                     }
+
+                    this.metrics.ON_PROBE_BASES   += onBaitBases;
+                    this.metrics.NEAR_PROBE_BASES += (mappedBases - onBaitBases);
+                }
+                else {
+                    this.metrics.OFF_PROBE_BASES += mappedBases;
                 }
             }
 
-            // Now do the bait overlaps
-            int mappedBases = 0;
-            for (final AlignmentBlock block : rec.getAlignmentBlocks()) mappedBases += TargetMetricsCollector.getNumBasesPassingMinimumBaseQuality(rec, block, minimumBaseQuality);
-            int onBaitBases = 0;
-
-            if (probes != null && !probes.isEmpty()) {
-                for (final Interval bait : probes) {
-                    for (final AlignmentBlock block : rec.getAlignmentBlocks()) {
-                        final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
-                        for (int pos=block.getReferenceStart(), readPos = block.getReadStart();
-                             pos<=end && pos >= bait.getStart() && pos <= bait.getEnd() && minimumBaseQuality <= baseQualities[readPos-1];
-                             ++pos, ++readPos) {
-                            ++onBaitBases;
+            ///////////////////////////////////////////////////////////////////
+            // And lastly, ignore reads falling below the mapq threshold
+            ///////////////////////////////////////////////////////////////////
+            if (this.mapQFilter.filterOut(record)) return;
```

Comment:
since this filter means that we should not consider the bases "to contribute to coverage", should this not move to before calculating `ON_PROBE_BASES`, `NEAR_PROBE_BASES`, `OFF_PROBE_BASES`?  Or are you considering those metrics not `coverage` metrics?